geth/node/node.go

```diff
@@ -123,16 +123,24 @@ func defaultEmbeddedNodeConfig(config *params.NodeConfig) *node.Config {
 			MaxPeers:         config.MaxPeers,
 			MaxPendingPeers:  config.MaxPendingPeers,
 		},
-		IPCPath:     makeIPCPath(config),
-		HTTPHost:    config.HTTPHost,
-		HTTPPort:    config.HTTPPort,
+		IPCPath: makeIPCPath(config),
+		// Disable this by default allow and set if given config allows http server.
+		// HTTPHost:    config.HTTPHost,
+		// HTTPPort:    config.HTTPPort,
 		HTTPCors:    []string{"*"},
 		HTTPModules: strings.Split(config.APIModules, ","),
 		WSHost:      makeWSHost(config),
 		WSPort:      config.WSPort,
 		WSOrigins:   []string{"*"},
 		WSModules:   strings.Split(config.APIModules, ","),
 	}
+
+	if config.HTTPEnabledMode {
```

Comment:
HTTPEnabledMode sounds a bit unclear to me. From one side, there's only 1 possible option implied by HTTPEnable and this is RPC. From the other side, HTTPEnabled is not the same as RPCEnabled and maybe we should consider using the second. What do you think?